share/scripts/tcl_bridge.tcl

```diff
@@ -0,0 +1,113 @@
+# tcl_bridge v1.0 (c) 2025 Pedro de Medeiros
+namespace eval tcl_bridge {
+
+variable init 0
+variable latch 0
+variable address
+variable result
+
+set_help_text tcl_bridge \
+{tcl_bridge -- allows guest/host communication through a MSX-DOS-to-Tcl bridge.
+
+Usage:
+  tcl_bridge start      Create communication channel
+  tcl_bridge stop       Destroy communication channel
+}
+
+proc tcl_bridge {args} {
+    if {[catch {set result [dispatcher {*}$args]} msg]} {
+        puts stderr $::errorInfo
+        error $msg
+    }
+    return $result
+}
+
+proc dispatcher {args} {
+    set params "[lrange $args 1 end]"
+    set cmd [lindex $args 0]
+    switch -- $cmd {
+        start   { return [tcl_bridge::start {*}$params] }
+        stop    { return [tcl_bridge::stop  {*}$params] }
+        default { error "Unknown command \"[lindex $args 0]\"." }
+    }
+}
+
+proc start {} {
+    variable init
+    if {$init eq 0} {
+        puts "tcl_bridge v1.0 (c) 2025 Pedro de Medeiros."
+        uplevel #0 {ext programmabledevice}
+        uplevel #0 {set {Programmable Device ports} {6 7}}
+        uplevel #0 {set {Programmable Device output callback} "tcl_bridge::read_command" }
+        uplevel #0 {set {Programmable Device input callback}  "tcl_bridge::write_command"}
+        uplevel #0 {set {Programmable Device reset callback}  "tcl_bridge::reset_command"}
+        set init 1
+        puts "tcl_bridge initialized"
+    } else {
+        puts "tcl_bridge already running"
+    }
+}
+
+proc stop {} {
+    variable init
+    if {$init eq 1} {
+        remove_extension programmabledevice
+        puts "tcl_bridge stopped"
+    } else {
+        puts "tcl_bridge not running"
+    }
+}
+
+proc read_string {base} {
+    set addr $base
+    set s ""
+    for {set c {}} {[set c [peek $addr]] ne 0} { incr addr } {
+        append s [format %c $c]
+        # detect overflows and runaways
+        if {$addr < $base || $addr > $base + 256} { break }
```

Comment:
This does always peek 256 bytes, right?
Maybe stop at the first `0` character? (And for safety still refuse to read more than 256 chars).